File: WebGl/static/color.js
class ColorRGBA {
    r = 1
    g = 1
    b = 1
    a = 0

    constructor(r, g, b, a) {
        this.r = r
        this.g = g
        this.b = b
        if (typeof a === "undefined")
            this.a = 1
        else
            this.a = a
    }

    getRGBA() {
        return [this.r, this.g, this.b, this.a]
    }
}

export class RGBA {
    static WHITE = new ColorRGBA(1., 1., 1.);
    static RED = new ColorRGBA(1., 0., 0.);
    static GREEN = new ColorRGBA(0., 1., 0.);
    static BLUE = new ColorRGBA(0., 0., 1.);
    // YELLOW = new RGB = kRedRGBA + kGreenRGBA;
    // PURPLE = new RGB = kRedRGBA + kBlueRGBA;
    // WHITE = new RGB = kRedRGBA + kGreenRGBA + kBlueRGBA;
}

File: C++/OpenGL/src/color.cpp
#include "headers/color.h"

color::RGB color::RGB::operator+(const RGB& other) const
{
    return RGB(this->red + other.red, this->green + other.green, this->blue + other.blue);
}

color::RGBA color::RGBA::operator+(const RGBA& other) const
{
    return RGBA{ this->red + other.red, this->green + other.green, this->blue + other.blue, this->alfa + other.alfa };
}